app/views/partials/paymentAllocations/PaymentAllocationsForNonLpi.scala.html

```diff
@@ -27,9 +26,9 @@
 
 @paymentAllocationHref(paymentAllocation: AllocationDetailWithClearingDate, isAgent: Boolean) = @{
     if(isAgent){
-        controllers.routes.ChargeSummaryController.showAgent(paymentAllocation.allocationDetail.getOrElse(throw MissingFieldException("Payment Allocation Detail")).getTaxYear(implicitDateFormatter), paymentAllocation.allocationDetail.getOrElse(throw MissingFieldException("Payment Allocation Detail")).transactionId.getOrElse(throw MissingFieldException("Document ID")))
+        controllers.routes.ChargeSummaryController.showAgent(paymentAllocation.validatedAllocationDetail.getTaxYear(implicitDateFormatter), paymentAllocation.validatedAllocationDetail.validatedTransactionId)
```

Comment:
assuming we are going to introduce viewModel here, calling validation will not make sense